### ISSUE TITLE
Advanced monitoring of model training

### ISSUE THREAD

**Issue by MaxenV** [MaxenV/Instruments-Classification#18]
To enhance training visibility, add a TensorBoard callback to the model training process, allowing real-time monitoring of metrics such as loss and accuracy. Users will be able to start a TensorBoard server to view detailed, live graphs of model performance across epochs, even if the notebook is disconnected or reloaded. This approach provides immediate insights into model behavior, helping with debugging and fine-tuning.

```python
from tensorflow.keras.callbacks import TensorBoard
import time

log_dir = f"logs/fit/{time.strftime('%Y%m%d-%H%M%S')}"
tensorboard_callback = TensorBoard(log_dir=log_dir, histogram_freq=1)

# Użyj callbacku podczas trenowania modelu
model.fit(..., callbacks=[tensorboard_callback])
Następnie uruchom TensorBoard w terminalu:
```
```bash
tensorboard --logdir logs/fit
```
Then access `http://localhost:6006` in a browser to monitor training metrics.
